src/colors/ui.ts

```diff
@@ -74,6 +74,9 @@ const ui = {
   white: {
     500: '#FFFFFF',
   },
+  black: {
```

Comment:
So i would keep this AND in the `theme.ts` where we generate the palette. 

You should add `defaultColors` to the palette this way:
```javascript
{
   defaultColors: {
      black: 500,
      blue: 300,
   }
}
```
and THEN, you need to adjust the palette type in `augmentation.ts` to accommodate for `defaultColors`. Make the colors type in `defaultColors` the explicit, non-optional, and always strings.